# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/SharedSurfaceGLX.h
# Commit: bbfe23c61add
# Full Hash: bbfe23c61add07377daf80696fd6996c0463f5b6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 8fcfac450ecc -r bbfe23c61add gfx/gl/SharedSurfaceGLX.h
--- a/gfx/gl/SharedSurfaceGLX.h	Mon Jun 08 20:13:01 2020 +0000
+++ b/gfx/gl/SharedSurfaceGLX.h	Mon Jun 08 20:34:15 2020 +0000
@@ -14,13 +14,18 @@
 namespace mozilla {
 namespace gl {
 
-class SharedSurface_GLXDrawable : public SharedSurface {
+class SharedSurface_GLXDrawable final : public SharedSurface {
  public:
-  static UniquePtr<SharedSurface_GLXDrawable> Create(GLContext* prodGL,
-                                                     const SurfaceCaps& caps,
-                                                     const gfx::IntSize& size,
-                                                     bool deallocateClient,
-                                                     bool inSameProcess);
+  const RefPtr<gfxXlibSurface> mXlibSurface;
+
+  static UniquePtr<SharedSurface_GLXDrawable> Create(const SharedSurfaceDesc&);
+
+ private:
+  SharedSurface_GLXDrawable(const SharedSurfaceDesc&,
+                            const RefPtr<gfxXlibSurface>& xlibSurface);
+
+ public:
+  ~SharedSurface_GLXDrawable();
 
   virtual void ProducerAcquireImpl() override {}
   virtual void ProducerReleaseImpl() override;
@@ -28,37 +33,18 @@
   virtual void LockProdImpl() override;
   virtual void UnlockProdImpl() override;
 
-  virtual bool ToSurfaceDescriptor(
-      layers::SurfaceDescriptor* const out_descriptor) override;
+  Maybe<layers::SurfaceDescriptor> ToSurfaceDescriptor() override;
 
   virtual bool ReadbackBySharedHandle(
       gfx::DataSourceSurface* out_surface) override;
-
- private:
-  SharedSurface_GLXDrawable(GLContext* gl, const gfx::IntSize& size,
-                            bool inSameProcess,
-                            const RefPtr<gfxXlibSurface>& xlibSurface);
-
-  RefPtr<gfxXlibSurface> mXlibSurface;
-  bool mInSameProcess;
 };
 
-class SurfaceFactory_GLXDrawable : public SurfaceFactory {
+class SurfaceFactory_GLXDrawable final : public SurfaceFactory {
  public:
-  static UniquePtr<SurfaceFactory_GLXDrawable> Create(
-      GLContext* prodGL, const SurfaceCaps& caps,
-      const RefPtr<layers::LayersIPCChannel>& allocator,
-      const layers::TextureFlags& flags);
+  explicit SurfaceFactory_GLXDrawable(GLContext&);
 
-  virtual UniquePtr<SharedSurface> CreateShared(
-      const gfx::IntSize& size) override;
-
- private:
-  SurfaceFactory_GLXDrawable(GLContext* prodGL, const SurfaceCaps& caps,
-                             const RefPtr<layers::LayersIPCChannel>& allocator,
-                             const layers::TextureFlags& flags)
-      : SurfaceFactory(SharedSurfaceType::GLXDrawable, prodGL, caps, allocator,
-                       flags) {}
+  virtual UniquePtr<SharedSurface> CreateSharedImpl(
+      const SharedSurfaceDesc& desc) override;
 };
 
 }  // namespace gl